src/app/customers/customers.component.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, OnInit } from '@angular/core';
+import axios from 'axios';
+import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
+import { CustomerDetailComponent } from '../customer-detail/customer-detail.component';
+import { ConfirmDialogComponent } from '../confirm-dialog/confirm-dialog.component';
+import { AddCustomerDialogComponent } from '../add-customer-dialog/add-customer-dialog.component';
+import { Customer } from './customer';
+import { environment } from 'src/environments/environment';
+
+
+interface CustomerContract
+{
+  UserID: number,
+   RoleID: number,
+   first_name: string,
+   last_name: string,
+   ContractID: number
+}
+interface CustomerExtend
+{
+  UserID: number,
+  RoleID: number,
+  first_name: string,
+  last_name: string,
+  BirthDate: Date,
+  AddressID: number,
+  Email: string,
+  PhoneNumber: string,
+  Password: string,
+  ContractID: number
+}
+
+@Component({
+  selector: 'app-customers',
+  templateUrl: './customers.component.html',
+  styleUrls: ['./customers.component.css']
+
+})
+export class CustomerComponent implements OnInit {
+ 
+  clientsArray!:CustomerContract[];
```

Comment:
clients describes it would be an array already. Watch out with using !. Initialize this properly. You can just make it an empty array `= []`. This applies to other places as well.